### PR TITLE
Test volume.block.filesystem ignore

### PR DESCRIPTION
**Pull request by masnax**
Related to https://github.com/canonical/lxd/pull/12938

### CODE REVIEW

**Comment by masnax** [comment]
So `nsenter -m -t "${serverPID}" stat -f -c %T /var/snap/lxd/common/lxd/virtual-machines/v1` actually outputs `ext2/ext3` when `volume.block.filesystem=ext4`

File: tests/storage-vm [hunk 1]
```diff
@@ -317,17 +317,20 @@ for poolDriver in $poolDriverList; do
...
                 lxc info v1
 
-                echo "==> Checking VM config disk filesyste is XFS"
+                # Ensure the VM volume ignores volume.block.filesystem.
+                lxc storage volume get "${poolName}" virtual-machine/v1 block.filesystem | grep -q "ext4"
+
+                echo "==> Checking VM config disk filesystem is not XFS"
                 serverPID="$(lxc query /1.0 | jq .environment.server_pid)"
-                nsenter -m -t "${serverPID}" stat -f -c %T /var/snap/lxd/common/lxd/virtual-machines/v1 | grep -xF xfs
+                nsenter -m -t "${serverPID}" stat -f -c %T /var/snap/lxd/common/lxd/virtual-machines/v1 | grep -xF ext2/ext3
```
Comment (simondeziel) [inline tests/storage-vm:333]:
```suggestion
                nsenter -m -t "${serverPID}" findmnt --noheadings --output=FSTYPE --mountpoint /var/snap/lxd/common/lxd/devices/v1/config.mount | grep -xF ext4
```
Maybe?

File: tests/storage-vm [hunk 2]
```diff
@@ -317,17 +317,20 @@ for poolDriver in $poolDriverList; do
...
                 echo "==> Change volume.block.filesystem on pool and create VM"
                 lxc storage set "${poolName}" volume.block.filesystem xfs
                 lxc init "${TEST_IMG:-ubuntu-minimal-daily:22.04}" v1 --vm -s "${poolName}"
                 lxc start v1
                 waitInstanceReady v1
                 lxc info v1
 
-                echo "==> Checking VM config disk filesyste is XFS"
+                # Ensure the VM volume ignores volume.block.filesystem.
+                lxc storage volume get "${poolName}" virtual-machine/v1 block.filesystem | grep -q "ext4"
```
Comment (simondeziel) [inline tests/storage-vm:329]:
```suggestion
                echo "==> Ensure the VM volume ignores volume.block.filesystem"
                volume_fs="$(lxc storage volume get "${poolName}" virtual-machine/v1 block.filesystem)"
                echo "${volume_fs}" | grep -F "ext4"
```
Maybe the above would help figure why it failed in https://github.com/canonical/lxd-ci/actions/runs/8012265928/job/21887207157?pr=80#step:6:1775
Removing `-q` as

Comment (masnax) [reply tests/storage-vm:329]: Hasn't been backported yet. https://github.com/canonical/lxd/pull/12950

Comment (masnax) [reply tests/storage-vm:329]: But good to add.

Comment (simondeziel) [reply tests/storage-vm:329]: Of course it's not yet in 5.0/edge, face palm.

In the suggested change, I also converted the comment into an echo as I find them useful, especially in long scripts like `tests/storage-vm`.

Comment (masnax) [reply tests/storage-vm:329]: > In the suggested change, I also converted the comment into an echo as I find them useful, especially in long scripts like `tests/storage-vm`.

Yeah I agree, I've been trying to do that with the microcloud tests as well.

**Review by tomponline — APPROVED** [review]
Thanks!

**Pull request merged**